btclib/psbt/psbt.py

```diff
@@ -452,3 +465,74 @@ def extract_tx(psbt: Psbt, check_validity: bool = True) -> Tx:
     if check_validity:
         tx.assert_valid()
     return tx
+
+
+def _sort_or_shuffle_together(
+    sequence_a: Sequence[Any],
+    sequence_b: Sequence[Any],
+    ordering_func: Callable[[Any], int] | None = None,
+) -> tuple[list[Any], list[Any]]:
+
+    if len(sequence_a) != len(sequence_b):
+        raise BTClibValueError("sequences must have same length")
+
+    tmp = list(zip(sequence_a, sequence_b))
+    if ordering_func is None:
+        random.shuffle(tmp)
+    else:
+        tmp.sort(key=lambda t: ordering_func(t[0]))  # type: ignore
+    tuple_a, tuple_b = zip(*tmp)
+    return list(tuple_a), list(tuple_b)
+
+
+def _ensure_consistency(psbts: Sequence[Psbt]) -> None:
+    if any(
+        k in psbts[0].hd_key_paths and v != psbts[0].hd_key_paths[k]
+        for psbt in psbts
+        for k, v in psbt.hd_key_paths.items()
+    ):
+        raise BTClibValueError("inconsistent hd_key_paths")
```

Comment:
Should this really be an exception? Maybe just a warning?

Reply:
Maybe I'm just being too pedantic, but it just doesn't make any sense for a pubkey to be associated to more than one (fingerprint, derivation_path) couple, does it? In fact from my point of view that should be a probabilistic bijection.

Reply:
resolved: the check is necessary